rest/model/data/meta.go

```diff
@@ -442,6 +458,33 @@ func validateNoteLength(v reflect.Value) error {
 	return nil
 }
 
+func validatePulsar(v reflect.Value) error {
+	var pulsars []*PulsarMeta
+
+	if v.Kind() == reflect.Slice {
+		// Slice from API
+		bs, err := json.Marshal(v.Interface())
+		if err != nil {
+			return fmt.Errorf("pulsar: unexpected value: `%v`", v.Interface())
+		}
+		if err := json.Unmarshal(bs, &pulsars); err != nil {
+			return fmt.Errorf("pulsar: invalid value: `%v`", v.Interface())
+		}
+	} else if v.Kind() == reflect.String {
+		// String from terraform
+		if err := json.Unmarshal([]byte(v.String()), &pulsars); err != nil {
+			return fmt.Errorf("pulsar: invalid value: `%v`", v.String())
```

Comment:
This validates that correct values were passed, but will ignore any fields not present in `PulsarMeta`.  That means a pulsar string passed from TF with invalid fields will pass validation and make it into the API request.  Ideally we should catch that here, no?

Reply:
point, but I'm inclined to leave it this way (for now at least) and just ensure we have the required field set. API will reject extra keys (but not in plan stage) and leaving some flexibility could be nice since these field options are not as in sync with documentation as they could be in general.

Reply:
👍Totally, as is it fixes the bug and is obvi a huge improvement over what was there.